### PR TITLE
Refactor samples, pantry etcetera

### PR DESCRIPTION
**Pull request by trexfeathers**
### Description

---

Applying some lessons learned from introducing GeoVista's 'assets' structure to new developers - seeing what was easy to grasp, and what was hard.

**If we like this: corresponding changes will be needed in the structure of [geovista-data](https://github.com/bjlittle/geovista-data).**

[Please take a look at the new structure in my branch, to get a feel for it.](https://github.com/trexfeathers/geovista/tree/refactor_pantry/src/geovista)

- Made [`geovista.cache`](https://github.com/trexfeathers/geovista/tree/refactor_pantry/src/geovista/cache) into a package directory, allowing the inclusion of [`registry.txt`](https://github.com/trexfeathers/geovista/blob/refactor_pantry/src/geovista/cache/registry.txt) and an explanatory [`README.md`](https://github.com/trexfeathers/geovista/blob/refactor_pantry/src/geovista/cache/README.md).
- Made a `geovista.pantry` package directory, encompassing several sorts of reusable things that were previously scattered through several root modules:
  - [`fetch_coastlines()`](https://github.com/trexfeathers/geovista/blob/refactor_pantry/src/geovista/pantry/__init__.py#L26) has moved to the `geovista.pantry` root
  - The previous ~`geovista.pantry`~ has become [`geovista.pantry.data`](https://github.com/trexfeathers/geovista/blob/refactor_pantry/src/geovista/pantry/data.py).
  - ~`geovista.samples`~ has become [`geovista.pantry.meshes`](https://github.com/trexfeathers/geovista/blob/refactor_pantry/src/geovista/pantry/meshes.py).
  - The texture routines previously in ~`geovista.cache`~ have moved to [`geovista.pantry.textures`](https://github.com/trexfeathers/geovista/blob/refactor_pantry/src/geovista/pantry/textures.py).

Hopefully I've caught everything - refactoring can be rather fraught!

### CODE REVIEW

File: src/geovista/__init__.py [hunk 1]
```diff
@@ -26,19 +26,18 @@
 import os
 
 from .bridge import Transform  # noqa: F401
-from .cache import (  # noqa: F401
-    blue_marble,
-    checkerboard,
-    fetch_coastlines,
```
Comment (trexfeathers) [inline src/geovista/__init__.py:32]: Whoops, didn't mean to remove `fetch_coastlines()`!

File: src/geovista/pantry/data.py [hunk 2]
```diff
@@ -24,8 +24,8 @@
 from numpy import ma
 import pooch
 
-from .cache import CACHE
-from .common import LRU_CACHE_SIZE, _MixinStrEnum
+from geovista.cache import CACHE
```
Comment (trexfeathers) [inline src/geovista/pantry/data.py:27]: Note that one of the pre-commit hooks (can't remember which one) objects to `from ..cache`, necessitating this change now that the module has moved into a subdirectory

File: src/geovista/__init__.py (hunk 1 above)
Comment (trexfeathers) [reply src/geovista/__init__.py:32]: 4b3abaca6dfa33c09fa75dfe1d347df83af4ee31

**Review by bjlittle — APPROVED** [review]
@trexfeathers Thanks, this is awesome 💯🚀  

It's a lovely enhancement that makes the code way more coherent and structured. This will make things a lot more obvious/clearer to other `geovista` developers 👍

**Comment by trexfeathers** [comment]
Thanks @bjlittle!

**Comment by bjlittle** [comment]
@all-contributors please add @trexfeathers for maintenance

**Pull request merged**